pkg/wait/wait.go

```diff
@@ -0,0 +1,89 @@
+package wait
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/submariner-io/armada/pkg/defaults"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ForPodsRunning waits for pods to be running
+func ForPodsRunning(clName string, c kubernetes.Interface, namespace, selector string, replicas int) error {
+	ctx := context.Background()
+	log.Debugf("Waiting for pods to be running. label: %q, namespace: %q, replicas: %v, duration: %v, *types: %s.", selector, namespace, replicas, defaults.WaitDurationResources, clName)
+	podsContext, cancel := context.WithTimeout(ctx, defaults.WaitDurationResources)
+	wait.Until(func() {
+		podList, err := c.CoreV1().Pods(namespace).List(metav1.ListOptions{
+			LabelSelector: selector,
+			FieldSelector: "status.phase=Running",
+		})
+		if err == nil && len(podList.Items) == replicas {
+			log.Infof("✔ All pods with label: %q are running for %s.", selector, clName)
+			cancel()
+		} else {
+			log.Debugf("Still waiting for pods. label: %q, namespace: %q, replicas: %v, cluster: %s.", selector, namespace, replicas, clName)
+		}
+	}, 15*time.Second, podsContext.Done())
```

Comment:
It seems like we should be able to check more frequently then 15 sec, say every 5 sec. Same on L58.

Reply:
Done w/ https://github.com/submariner-io/armada/pull/1